.github/workflows/ci-cd.yml

```diff
@@ -43,8 +43,21 @@ jobs:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
       
+      - name: Build versioned docker image
+        run: docker build --target production --tag ${{ secrets.DOCKER_USERNAME }}/todo-app:$GITHUB_SHA .
+
       - name: Build prod docker image
         run: docker build --target production --tag ${{ secrets.DOCKER_USERNAME }}/todo-app:prod .
       
-      - name: Push prod docker image to Docker Hub
+      - name: Push versioned image to Docker Hub
+        run: docker push ${{ secrets.DOCKER_USERNAME }}/todo-app:$GITHUB_SHA
+      
+      - name: Push prod image to Docker Hub
         run: docker push ${{ secrets.DOCKER_USERNAME }}/todo-app:prod
+
+      - run: echo "Pushed versioned image with tag ${{ secrets.DOCKER_USERNAME }}/todo-app:$GITHUB_SHA and production image to Docker"
+
+      - name: Deploy updated image to Azure
+        run: curl -dH -X POST ${{ secrets.DEPLOY_WEBHOOK }}
```

Comment:
One nice addition here can be adding the [--fail](https://curl.se/docs/manpage.html#-f) flag, as otherwise curl reports success as long as a successful connection was made, even if the response reports a failure (e.g. a status code > 400) which can lead to confusing pipeline results